### DISCUSSION TITLE
How to turn on device with node-red component?

### DISCUSSION THREAD

**Discussion by bzq-pl** [alexkn/node-red-contrib-homeconnect#59] [Q&A] [answered]
Hi,

I have a dishwasher and when device is off then it's not possible to start program. Error message is "HomeConnectError: SDK.Error.WrongOperationState: Request cannot be performed since OperationState is not Ready”. most probably to turn on device I should call this:
curl -X PUT [https://api.home-connect.com/api/homeappliances/{haId}/settings/BSH.Common.Setting.PowerState](https://api.home-connect.com/api/homeappliances/%7BhaId%7D/settings/BSH.Common.Setting.PowerState) \
  -H "Authorization: Bearer {access_token}" \
  -H "Content-Type: application/vnd.bsh.sdk.v1+json" \
  --data-raw '{
   "data": {
    "key": "BSH.Common.Setting.PowerState",
    "value": "BSH.Common.EnumType.PowerState.On"
   }
  }'
but in this requires to pass access_code

Do you know how can I turn on device via node-red?

Regards
Piotr

**Comment by alexkn** [comment, marked as answer]
You should be able to send such command with the home-connect-request node:
Tag: `Settings`
Action: `set_setting`
Setting key: `BSH.Common.Setting.PowerState`
HAID:
Body:
```json
{
  "data": {
    "key": "BSH.Common.Setting.PowerState",
    "value": "BSH.Common.EnumType.PowerState.On"
  }
}
```

~~But I don't think you can start a program this way.~~
You also need the [Remote Start Allowance State](https://api-docs.home-connect.com/states?#remote-start-allowance-state).

**Reply by bzq-pl** [reply]
This works. I have enabled earlier remote start so ... if doors are closed then I can remotely start program :-)

Thank you!!!
